sim/rogue/combat/blade_flurry.go

```diff
@@ -0,0 +1,78 @@
+package combat
+
+import (
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+)
+
+var BladeFlurryActionID = core.ActionID{SpellID: 13877}
+var BladeFlurryHitID = core.ActionID{SpellID: 22482}
+
+func (comRogue *CombatRogue) registerBladeFlurry() {
+	var curDmg float64
+	bfHit := comRogue.RegisterSpell(core.SpellConfig{
+		ActionID:    BladeFlurryHitID,
+		SpellSchool: core.SpellSchoolPhysical,
+		ProcMask:    core.ProcMaskEmpty, // No proc mask, so it won't proc itself.
+		Flags:       core.SpellFlagMeleeMetrics | core.SpellFlagNoOnCastComplete | core.SpellFlagIgnoreAttackerModifiers,
+
+		DamageMultiplier: 1,
+		ThreatMultiplier: 1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			spell.CalcAndDealDamage(sim, target, curDmg, spell.OutcomeAlwaysHit)
+		},
+	})
+
+	hasGlyph := comRogue.HasMajorGlyph(proto.RogueMajorGlyph_GlyphOfBladeFlurry)
+	energyReduction := core.TernaryFloat64(hasGlyph, -0.15, -0.3)
+
+	comRogue.BladeFlurryAura = comRogue.RegisterAura(core.Aura{
+		Label:    "Blade Flurry",
+		ActionID: BladeFlurryActionID,
+		Duration: core.NeverExpires,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			comRogue.ResetEnergyTick(sim)
+			comRogue.ApplyEnergyTickMultiplier(energyReduction)
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			comRogue.ResetEnergyTick(sim)
+			comRogue.ApplyEnergyTickMultiplier(-energyReduction)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if sim.GetNumTargets() < 2 {
+				return
+			}
+			if result.Damage == 0 || !spell.ProcMask.Matches(core.ProcMaskMelee) {
+				return
+			}
+			// Fan of Knives is not cloned
+			if spell.IsSpellAction(comRogue.FanOfKnives.SpellID) {
+				return
+			}
+
+			// Undo armor reduction to get the raw damage value.
+			curDmg = result.Damage / result.ResistanceMultiplier
```

Comment:
Is the BF hit supposed to be an exact clone of the main hit? Isnt it just an extra attack on the cleave target?

Reply:
Yep, exact clone of the original hit. There's some tech around hitting the lowest armor targets to gain damage.